src/GitHub.Api/ApiClientConfiguration_User.cs

```diff
@@ -1,11 +1,13 @@
+﻿using System;
+
 namespace GitHub.Api
 {
-    public partial class ApiClient : IApiClient
+    static partial class ApiClientConfiguration
```

Comment:
We might need to tweak the csproj and the default codepath a bit so that if it doesn't find the file, it can still build with fake data (right now the file always has to be there). Probably also better to move it to the `common` folder, since it's "common" configuration data and that's an easier location to point to than all the way inside GitHub.Api, and maybe rename it to something more obvious. The steps would probably be something like:

- Set default `ClientId` and `ClientSecret` values in `ApiClientConfiguration.cs` if they're null after `Configure()`
- Delete the public `ApiClientConfiguration_User.cs` file
- Add a `src/common/ApplicationKeys.cs` file to .gitignore
- Include `src/common/ApplicationKeys.cs` in the csproj if it exists and it's not an `internal` build
- Add a little script that asks the user for a client id and client secret and generates the code with them.
- Update the docs with these instructions (and also the template if people want to do it manually)

We can do this on a separate PR, probably 😄